planner/core/planbuilder.go

```diff
@@ -697,9 +702,10 @@ func (b *PlanBuilder) getPossibleAccessPaths(indexHints []*ast.IndexHint, tbl ta
 	// Extract comment-style index hint like /*+ INDEX(t, idx1, idx2) */.
 	indexHintsLen := len(indexHints)
 	if hints := b.TableHints(); hints != nil {
-		for _, hint := range hints.indexHintList {
+		for i, hint := range hints.indexHintList {
 			if hint.dbName.L == dbName.L && hint.tblName.L == tblName.L {
 				indexHints = append(indexHints, hint.indexHint)
+				hints.indexHintList[i].matched = true
```

Comment:
What if `use_index(table_name)`, none index name. It means we should choose indexes for the table, but it doesn't match any index name.

Reply:
IMO, `use_index(table_name)` means choose `TableScan`.